clients/src/main/java/org/apache/kafka/clients/consumer/internals/DefaultBackgroundThread.java

```diff
@@ -168,28 +171,41 @@ public DefaultBackgroundThread(final Time time,
                             logContext);
             CoordinatorRequestManager coordinatorRequestManager = null;
             CommitRequestManager commitRequestManager = null;
+            HeartbeatRequestManager heartbeatRequestManager = null;
+            MembershipManager membershipManaber = null;
```

Comment:
This is only needed when there is a groupId defined so I would move it completely to the `if (groupState.groupId != null)` block